ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala

```diff
@@ -107,32 +119,97 @@ class FilterTableACSReader(
       .map(toIdSource)
       .pipe(mergeSort[Long])
       .statefulMapConcat(statefulDeduplicate)
-      .grouped(pageSize)
-      .map(_.toVector)
+      .via(
+        BatchN(
```

Comment:
by switching from grouped to dynamic BatchN batching we get size=1 batches first, which quickly increasing to the mazimum batch size.
This gives us very fast stream initialization followed by uniform maximum batch formation after.
Experiments show the uniform-maximum batch formation is still the case, if there is backpressure from the consumer.